[BUG] new() generic constraint is lost in generated client

**Describe the bug**
new() generic constraint is not handled properly in generated client

The original interface:
```csharp
public class DatasetQueryItem<TResultRow>
        where TResultRow : class, new()
    {
        [JsonProperty("global_id")]
        public long GlobalId { get; set; }

        public long Number { get; set; }

        [JsonProperty("Cells")]
        public TResultRow Value { get; set; }
    }

public interface IDataMosApi
    {
        [Get("/datasets/{dataSet}/rows")]
        Task<DatasetQueryItem<TResulRow>[]> GetDataSetItems<TResulRow>()          
            where TResulRow : class, new(); // <========== here
    }
```

Generated client:
```csharp
public global::System.Threading.Tasks.Task<global::Pp3.Integration.Shared.DataMos.DatasetQueryItem<TResulRow>[]> GetDataSetItems<TResulRow>() 
         where TResulRow : class // <========== new() is lost
```

Build props:
- `<LangVersion>latest</LangVersion>`
- .net core 5.0.4
- SDK 5.0.201
- Refit 6.0.24

**Actual behavior**

```
InterfaceStubGenerator.Core\Refit.Generator.InterfaceStubGenerator\IDataMosApi.g.cs(33,122): error CS0310: 'TResulRow' must be a non-abstract type with a public parameterless constructor in order to use it as parameter 'TResultRow' in the generic type or method 'DatasetQueryItem<TResultRow>'
```

**Expected behavior**
Successful build

**Environment**
- OS:  Windows 10
- Refit 6.0.24

--- a/InterfaceStubGenerator.Core/InterfaceStubGenerator.cs
+++ b/InterfaceStubGenerator.Core/InterfaceStubGenerator.cs
@@ -503,7 +503,7 @@ namespace Refit.Implementation
                 source.Append(string.Join(", ", list));
             }
             
-           source.Append(@$") {GenerateConstraints(methodSymbol.TypeParameters, true)}
+           source.Append(@$") {GenerateConstraints(methodSymbol.TypeParameters, isExplicitInterface)}
         {{");
         }
 

--- a/Refit.Tests/InheritedGenericInterfacesApi.cs
+++ b/Refit.Tests/InheritedGenericInterfacesApi.cs
@@ -1,5 +1,7 @@
 ﻿using System.Collections.Generic;
 using System.Threading.Tasks;
+using Newtonsoft.Json;
+
 using Refit; // InterfaceStubGenerator looks for this
 using static System.Math; // This is here to verify https://github.com/reactiveui/refit/issues/283
 
@@ -47,5 +49,29 @@ namespace Refit.Tests
 
         [Delete("/{key}")]
         Task Delete(TKey key);
+
+        [Get("")]
+        Task ReadAllClasses<TFoo>()
+            where TFoo : class, new();
+    }
+
+
+    public class DatasetQueryItem<TResultRow>
+        where TResultRow : class, new()
+    {
+        [JsonProperty("global_id")]
+        public long GlobalId { get; set; }
+
+        public long Number { get; set; }
+
+        [JsonProperty("Cells")]
+        public TResultRow Value { get; set; }
+    }
+
+    public interface IDataMosApi
+    {
+        [Get("/datasets/{dataSet}/rows")]
+        Task<DatasetQueryItem<TResulRow>[]> GetDataSetItems<TResulRow>()
+            where TResulRow : class, new(); 
     }
 }

